aten/src/ATen/Context.h

```diff
@@ -214,22 +214,22 @@ static inline bool hasMKLDNN() {
 }
 
 static inline void manual_seed(uint64_t seed) {
-  auto& gen = globalContext().defaultGenerator(DeviceType::CPU);
+  auto gen = globalContext().defaultGenerator(DeviceType::CPU);
   {
     // See Note [Acquire lock when using random generators]
-    std::lock_guard<std::mutex> lock(gen.mutex_);
-    gen.set_current_seed(seed);
+    std::lock_guard<std::mutex> lock(gen->mutex_);
+    gen->set_current_seed(seed);
   }
   // NB: Sometimes we build with CUDA, but we don't have any GPUs
   // available. In that case, we must not seed CUDA; it will fail!
   int num_gpus = detail::getCUDAHooks().getNumGPUs();
   if (hasCUDA() && num_gpus > 0) {
     for (int i = 0; i < num_gpus; i++) {
-      auto& cuda_gen = globalContext().defaultGenerator(Device(at::kCUDA, i));
+      auto cuda_gen = globalContext().defaultGenerator(Device(at::kCUDA, i));
       {
         // See Note [Acquire lock when using random generators]
-        std::lock_guard<std::mutex> lock(cuda_gen.mutex_);
-        cuda_gen.set_current_seed(seed);
+        std::lock_guard<std::mutex> lock(cuda_gen->mutex_);
+        cuda_gen->set_current_seed(seed);
```

Comment:
Also doesn't have to be this PR, but we should consider adding methods like `set_current_seed` to Generator directly, so dot-syntax works on them (making Generator consistent with Tensor)

Reply:
will do in #34987